service.go

```diff
@@ -35,7 +38,17 @@ func NewDiagnosticsServer(logFile string) *DiagnosticsServer {
 }
 
 // SearchLog implements the DiagnosticsServer interface.
-func (d *DiagnosticsServer) SearchLog(req *pb.SearchLogRequest, stream pb.Diagnostics_SearchLogServer) error {
+func (d *DiagnosticsServer) SearchLog(req *pb.SearchLogRequest, stream pb.Diagnostics_SearchLogServer) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			buf := make([]byte, 4096)
+			stackSize := runtime.Stack(buf, false)
+			buf = buf[:stackSize]
+			err = fmt.Errorf(fmt.Sprintf("search log panic, %v, stack is %v", r, string(buf)))
+			log.Error(err.Error())
+		}
```

Comment:
Should we return `err` to the outside?

Reply:
The error already returns to the outside since line 47 assign value to `err`.